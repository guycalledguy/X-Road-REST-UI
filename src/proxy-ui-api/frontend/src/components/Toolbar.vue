<template>
  <v-toolbar dark color="primary" class="elevation-2">
    <v-img
      :src="require('../assets/xroad_logo_64.png')"
      height="64"
      width="128"
      max-height="64"
      max-width="128"
    ></v-img>
    <div class="separator"></div>
    <div class="white--text server-name">name of the server</div>
    <v-spacer></v-spacer>Username?
    <v-menu bottom left>
      <v-btn slot="activator" dark icon>
        <v-icon>fa-user-circle</v-icon>
      </v-btn>

      <v-list>
        <v-list-tile @click="logout">
          <v-list-tile-title>Logout</v-list-tile-title>
        </v-list-tile>

        <v-list-tile @click="demoLogout">
          <v-list-tile-title>Logout without UI</v-list-tile-title>
        </v-list-tile>
      </v-list>
    </v-menu>
    <div id="header-title">Security Server</div>
  </v-toolbar>
</template>

<script lang="ts">
import Vue from 'vue';

export default Vue.extend({
  name: 'toolbar',
  methods: {
    logout(): void {
      this.$store.dispatch('logout');
      this.$router.replace('/login');
    },
    demoLogout(): void {
      this.$store.dispatch('demoLogout');
    },
  },
});
</script>

<style lang="scss" scoped>
.server-name {
  font-size: 14px;
  margin: 20px;
}

.separator {
  width: 2px;
  height: 24px;
  margin-left: 6px;
  background-color: white;
}

#header-title {
  text-align: center;
  position: absolute;
  left: 0;
  right: 0;
  margin-left: auto;
  margin-right: auto;
  width: 190px;
  opacity: 0.2;
  color: #ffffff;
  font-size: 24px;
  font-weight: 500;
  line-height: 29px;

  @media only screen and (max-width: 920px) {
    display: none;
  }
}
</style>
