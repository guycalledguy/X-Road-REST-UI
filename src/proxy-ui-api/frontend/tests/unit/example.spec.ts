import { mount, createLocalVue } from '@vue/test-utils';
import { SilenceWarnHack } from './silenceWarnHack';
import VueRouter from 'vue-router';
import Vuetify from 'vuetify';
import Toolbar from '@/components/Toolbar.vue';

const silenceWarnHack = new SilenceWarnHack();

describe('Toolbar', () => {
  let wrapper: any;

  const routes = [
    { path: '/clients', name: 'clients' },
  ];

  const router = new VueRouter({ routes });
  let localVue = null;

  beforeEach(() => {
    silenceWarnHack.enable();
    localVue = createLocalVue();
    localVue.use(VueRouter);
    localVue.use(Vuetify);
    silenceWarnHack.disable();

    wrapper = mount(Toolbar, {
      localVue,
      router,
    });
  });

  test('is a Vue instance', () => {
    expect(wrapper.isVueInstance()).toBeTruthy();
  });

  // `it' and `expect's ready to go now.
  it('snapshot test', () => {
    expect(wrapper.element).toMatchSnapshot();
  });

});
