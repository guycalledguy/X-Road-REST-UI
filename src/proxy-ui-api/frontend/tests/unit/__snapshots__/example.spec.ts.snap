// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Toolbar snapshot test 1`] = `
<nav
  class="elevation-2 v-toolbar theme--dark primary"
  style="margin-top: 0px; padding-right: 0px; padding-left: 0px; transform: translateY(0px);"
>
  <div
    class="v-toolbar__content"
    style="height: 64px;"
  >
    <div
      class="v-responsive v-image"
      style="height: 64px; max-height: 64px; max-width: 128px; width: 128px;"
    >
      <div
        class="v-responsive__content"
      />
    </div>
     
    <div
      class="separator"
    />
     
    <div
      class="white--text server-name"
    >
      name of the server
    </div>
     
    <div
      class="spacer"
    />
    Username?
  
    <div
      class="v-menu v-menu--inline"
    >
      <div
        class="v-menu__activator"
      >
        <button
          class="v-btn v-btn--icon theme--dark"
          type="button"
        >
          <div
            class="v-btn__content"
          >
            <i
              aria-hidden="true"
              class="v-icon fa fa-user-circle theme--dark"
            />
          </div>
        </button>
      </div>
      <div
        class="v-menu__content theme--light "
        style="max-height: auto; min-width: 0px; max-width: auto; top: 12px; left: 0px; z-index: 0; display: none;"
      >
         
        <div
          class="v-list theme--light"
          role="list"
        >
          <div
            role="listitem"
          >
            <a
              class="v-list__tile v-list__tile--link theme--light"
            >
              <div
                class="v-list__tile__title"
              >
                Logout
              </div>
            </a>
          </div>
           
          <div
            role="listitem"
          >
            <a
              class="v-list__tile v-list__tile--link theme--light"
            >
              <div
                class="v-list__tile__title"
              >
                Logout without UI
              </div>
            </a>
          </div>
        </div>
      </div>
    </div>
     
    <div
      id="header-title"
    >
      Security Server
    </div>
  </div>
</nav>
`;
